### ISSUE TITLE
Not providing a width attribute throws error

### ISSUE THREAD

**Issue by tpinne** [sitegeist/sms-responsive-images#89]
TYPO3 10.4.21
sms_responsive_images: 2.1.0

I have the following use of the `sms:image` viewhelper.

```html
<sms:image
          image="{image.file}"
          alt="{image.alternative}"
          title="{image.title}"
          srcset="400, 600, 800, 1000"
          maxWidth="900"
        />
```

which throws this error

```
Oops, an error occurred!
Argument 2 passed to Sitegeist\ResponsiveImages\Utility\ResponsiveImagesUtility::generateSrcsetImages() 
must be of the type int, null given, called in 
/srv/nfs/.../public/typo3conf/ext/sms_responsive_images/Classes/Utility/ResponsiveImagesUtility.php on line 124
```

Debug output at https://github.com/sitegeist/sms-responsive-images/blob/7c9ee32f2ed2834cde6918eaa11d5d5c8761cd70/Classes/ViewHelpers/ImageViewHelper.php#L153

### That's `$image`
![vlV3Y](https://user-images.githubusercontent.com/1384345/142464724-47ddc35f-c384-4e64-b419-963c2c1d0e1f.png)

### This is `$processingInstructions`
![Bildschirmfoto 2021-11-18 um 18 15 56](https://user-images.githubusercontent.com/1384345/142465048-a2fdfbd1-fde3-4fec-91a3-c96005a0652a.png)

There is a width of `NULL` because the value comes from `$this->arguments['width']` which is obviously `NULL` because the attribute is not set (and it's not required).

And that seems to result in having also `NULL` on the `width` property on the fallback image object. Which in turn is `NULL` her https://github.com/sitegeist/sms-responsive-images/blob/7c9ee32f2ed2834cde6918eaa11d5d5c8761cd70/Classes/Utility/ResponsiveImagesUtility.php#L96 and then here https://github.com/sitegeist/sms-responsive-images/blob/7c9ee32f2ed2834cde6918eaa11d5d5c8761cd70/Classes/Utility/ResponsiveImagesUtility.php#L123-L130 where the error is thrown.

### This is `$fallbackImage`
![7guqn](https://user-images.githubusercontent.com/1384345/142465118-317cb614-9b08-4bf7-b93f-f16a8f40a3cf.png)

If I add a width attribute the error goes away and it renders fine.

```html
<sms:image
          image="{image.file}"
          alt="{image.alternative}"
          title="{image.title}"
          srcset="400, 600, 800, 1000"
          width="900"
          maxWidth="900"
        />
```

I don't know if it of any relevance, but I use this viewhelper inside of custom component created with https://github.com/sitegeist/fluid-components. Here is the full implementation for reference.

```html
<html xmlns:f="http://typo3.org/ns/TYPO3/CMS/Fluid/ViewHelpers"
      xmlns:fc="http://typo3.org/ns/SMS/FluidComponents/ViewHelpers"
      xmlns:sms="http://typo3.org/ns/Sitegeist/ResponsiveImages/ViewHelpers"
      xmlns:iwc="http://typo3.org/ns/Imos/ImosWebsite/Components"
      data-namespace-typo3-fluid="true">

<fc:component>
  <fc:param name="headline" type="string" />
  <fc:param name="headlineType" type="string" optional="1" />
  <fc:param name="image" type="Image" />
  <fc:param name="link" type="Typolink" />
  <fc:param name="linkLabel" type="string" optional="1" />
  <fc:param name="backgroundClass" type="string" optional="1" default="bg-white" />

  <fc:renderer>
    <div class="{backgroundClass} relative transform-gpu hover:scale-105 transition-transform">
      <div>
        <sms:image
          class="col-start-1 col-end-10 md:order-1 place-self-end"
          image="{image.file}"
          alt="{image.alternative}"
          title="{image.title}"
          srcset="400, 600, 800, 1000"
          maxWidth="900"
        />
        <div class="p-xs text-sm text-highlight-main">
          <iwc:atom.header header="{headline}" layout="{headlineType}" overwriteAppearanceClass="font-bold mb-xs" />
          <div class="border-b border-main flex justify-between">
            <iwc:atom.morelink link="{link}" label="{linkLabel}" class="pb-3xs leading-none stretched-link" />
            <i class="fa-imos-webicon_up-right"></i>
          </div>
        </div>
      </div>
    </div>
  </fc:renderer>
</fc:component>
</html>
```

**Comment by s2b** [comment]
Unfortunately, I can't reproduce this behavior in my system (10.4.20, with components). If I use the following processing instructions on an Image:

```
$processingInstructions = [
    'width' => null,
    'crop' => null,
    'maxWidth' => 900
];
```

The resulting ProcessedImage looks like this:

<img width="814" alt="Bildschirmfoto 2021-11-22 um 09 21 33" src="https://user-images.githubusercontent.com/458524/142827239-0ad90a04-12e1-4420-a394-1b384810cfcc.png">

Could you check in InstallTool if your image cropping works correctly?

**Comment by tpinne** [comment]
Oh, how I love those errors. On thursday I could reproduce it without any problems. And today I saw the error message once after I started looking into it again, after I read your reply. And after investigating the content element which caused the error it disappeared. For good. No way I'm able to reproduce it now. Even with other elements on other pages which produced the same error. Now nothing, whether in workspace or not or editor or admin.... 😡

I'll close it for now and hope it was some kind of hiccup last week.
